src/read/read_ref.rs

```diff
@@ -47,8 +47,8 @@ pub trait ReadRef<'a>: Clone + Copy {
     /// Does not include the delimiter.
     ///
     /// Returns an error if offset is out of bounds or the delimiter is
-    /// not found.
-    fn read_bytes_at_until(self, offset: u64, delimiter: u8) -> Result<&'a [u8]>;
+    /// not found in the range offset..upper_bound.
+    fn read_bytes_at_until(self, offset: u64, delimiter: u8, upper_bound: u64) -> Result<&'a [u8]>;
```

Comment:
Maybe a `Range<u64>` would be clearer, or at least place `upper_bound` after `offset`?